bindings/wasm/src/lib.rs

```diff
@@ -66,208 +61,564 @@ fn js_error<T: std::fmt::Debug>(e: T) -> JsValue {
     JsValue::from(format!("{:?}", e))
 }
 
-fn create_hash(bytes: &[i8]) -> Hash {
-    let mut array = [0; 243];
-    let bytes = &bytes[..array.len()];
-    array.copy_from_slice(bytes); 
-    Hash(array)
+fn create_hash_from_string(bytes: String) -> Result<Hash, JsValue> {
+    let hash = Hash::from_inner_unchecked(
+        TryteBuf::try_from_str(&bytes)
+            .map_err(js_error)?
+            .as_trits()
+            .encode(),
+    );
+    Ok(hash)
+}
+
+fn create_hash(bytes: JsValue) -> Result<Hash, JsValue> {
+    let bytes: String = bytes.into_serde().map_err(js_error)?;
+    create_hash_from_string(bytes)
 }
 
-#[wasm_bindgen]
-impl Client {
-    #[wasm_bindgen(constructor)]
-    pub fn new(uri: &str) -> Result<Client, JsValue> {
-        console_error_panic_hook::set_once();
-        let client = Client {
-            client: iota::Client::new(uri)
-                .map_err(|e| e.to_string())?
-        };
-        Ok(client)
+fn create_hash_array(bytes_vec: JsValue) -> Result<Vec<Hash>, JsValue> {
+    let hashes_vecs: Vec<String> = bytes_vec.into_serde().map_err(js_error)?;
+    let mut hashes = Vec::new();
+    for hash in hashes_vecs {
+        hashes.push(create_hash_from_string(hash)?);
     }
+    Ok(hashes)
+}
 
-    #[wasm_bindgen(js_name = "getNodeInfo")]
-    pub async fn get_node_info(self) -> Result<JsValue, JsValue> {
-        let node_info = self.client.get_node_info()
-            .await
-            .map_err(|e| JsValue::from(e.to_string()))?;
-        let res = response_to_js_value(node_info)?;
-        Ok(res)
+fn create_address(address: String) -> Result<Address, JsValue> {
```

Comment:
These functions are because those types are lack of serialize trait right? Could you add a TODO here to remind us to implement this later?

Reply:
actually these functions are here to convert a javascript value (e.g. a string) to its Rust type (e.g. the Address).

the code that I had to write only because some types doesn't implement Serialize are the `impl Serialize for T` on client/response.rs